code/datums/storage/storage.dm

```diff
@@ -1117,6 +1091,116 @@ GLOBAL_LIST_EMPTY(cached_storage_typecaches)
+
+	storage_sets[ui_style] = list(cells, corner1, corner2, corner3, corner4, rowjoin1, rowjoin2)
+
+/// Updates views of all objects in storage and stretches UI to appropriate size
+/datum/storage/proc/orient_storage()
+	var/adjusted_contents = real_location.contents.len
+	var/list/datum/numbered_display/numbered_contents
+	if(numerical_stacking)
+		numbered_contents = process_numerical_display()
+		adjusted_contents = numbered_contents.len
+
+	//if the ammount of contents reaches some multiplier of the final column (and its not the last slot), let the player view an additional row
+	var/additional_row = (!(adjusted_contents % screen_max_columns) && adjusted_contents < max_slots)
+
+	var/columns = clamp(max_slots, 1, screen_max_columns)
+	var/rows = clamp(CEILING(adjusted_contents / columns, 1) + additional_row, 1, screen_max_rows)
+
+	for (var/ui_theme in storage_sets)
+		var/list/atom/movable/screen/storage/ui_set = storage_sets[ui_theme]
+		ui_set[STORAGE_CELL_INDEX].screen_loc = "[screen_start_x]:[screen_pixel_x],[screen_start_y]:[screen_pixel_y] to [screen_start_x + columns - 1]:[screen_pixel_x],[screen_start_y + rows - 1]:[screen_pixel_y]"
+
+		ui_set[STORAGE_TOP_LEFT_CORNER_INDEX].screen_loc = "[screen_start_x + 1]:[screen_pixel_x],[screen_start_y]:[screen_pixel_y] to [screen_start_x + columns - 1]:[screen_pixel_x],[screen_start_y + rows - 1]:[screen_pixel_y]"
+		ui_set[STORAGE_TOP_RIGHT_CORNER_INDEX].screen_loc = "[screen_start_x]:[screen_pixel_x],[screen_start_y]:[screen_pixel_y] to [screen_start_x + max(0, columns - 2)]:[screen_pixel_x],[screen_start_y + rows - 1]:[screen_pixel_y]"
+		ui_set[STORAGE_BOTTOM_LEFT_CORNER_INDEX].screen_loc = "[screen_start_x + 1]:[screen_pixel_x],[screen_start_y]:[screen_pixel_y] to [screen_start_x + columns - 1]:[screen_pixel_x],[screen_start_y + rows - 1]:[screen_pixel_y]"
+		ui_set[STORAGE_BOTTOM_RIGHT_CORNER_INDEX].screen_loc = "[screen_start_x]:[screen_pixel_x],[screen_start_y]:[screen_pixel_y] to [screen_start_x + max(0, columns - 2)]:[screen_pixel_x],[screen_start_y + rows - 1]:[screen_pixel_y]"
+
+		ui_set[STORAGE_LEFT_ROWJOIN_INDEX].screen_loc = "[screen_start_x]:[screen_pixel_x],[screen_start_y]:[screen_pixel_y + 27] to [screen_start_x]:[screen_pixel_x],[screen_start_y + max(0, rows - 2)]:[screen_pixel_y + 27]"
+		ui_set[STORAGE_LEFT_ROWJOIN_INDEX].alpha = (rows > 1) * 255
+
+		ui_set[STORAGE_RIGHT_ROWJOIN_INDEX].screen_loc = "[screen_start_x + columns - 1]:[screen_pixel_x],[screen_start_y]:[screen_pixel_y + 27] to [screen_start_x + columns - 1]:[screen_pixel_x],[screen_start_y + max(0, rows - 2)]:[screen_pixel_y + 27]"
+		ui_set[STORAGE_RIGHT_ROWJOIN_INDEX].alpha = (rows > 1) * 255
+
+		storage_closers[ui_theme].screen_loc = "[screen_start_x + columns]:[screen_pixel_x - 5],[screen_start_y]:[screen_pixel_y]"
+
+	var/current_x = screen_start_x
+	var/current_y = screen_start_y
+	var/turf/our_turf = get_turf(real_location)
+
+	if(islist(numbered_contents))
+		for(var/type in numbered_contents)
```

Comment:
type is a bad var name to use because it is ALWAYS being shadowed. need a new one. also what is this? type of what?

Reply:
This is old code for rendering items, not written by me. Not sure to be honest, but I'll rename it.